src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -716,5 +760,148 @@ static bool IsPositionWithinDocument(Position position, SourceText sourceText)
                 return position.Line < sourceText.Lines.Count;
             }
         }
+
+        private static bool TryGetDocumentChanges(WorkspaceEdit workspaceEdit, [NotNullWhen(true)] out TextDocumentEdit[]? documentChanges)
```

Comment:
Ah all this magic is from the HTMLC# helpers right?

Reply:
Yeah, this is all copied from the other side of the fence, just the actual "RemapCore" method changed, to do what the other side does via LSP